### PR TITLE
add additional load and save cases to comply with Tree data

### PR DESCRIPTION
**Pull request by benxu3**
Added treeData and optionData to constructor for TextFileTranslator.

String[][] optionData contains String[] arrays which represent key-value pairs for selectedOptions in Tree. Let me know what this data represents such that I can create the relevant Trees properly in a follow-up gameCreationInteractor PR. This was separated out from String[][] treeData to comply with the ArrayList<ArrayList<String[]>> return type. I tried making an ArrayList<Object[]> to create ArrayLists of arbitrary depth but that implementation causes errors since Java does not match ArrayList<String[]> types to Object[]. (I thought ArrayLists were Objects and thus it may have worked but that appears to be wrong)

String[][] treeData contains [int currentTreeIndex, int currentPlayerIndex], Integer[] tradingStates, Integer[] auctionStates, Integer[] mainStates arrays. This complies with the ArrayList<ArrayList<String[]>> return type.

String[][] playerData contains String[] arrays which represent either a Player instance with its attributes or a Property instance which is owned by a Player. (We may want to change the way Property is saved such that instead of denoting the Player which owns it by an index in the Players[] array, we denote it using the Player's name instance attribute) I will handle the case of differentiating between the two in a follow-up gameCreationInteractor PR.

String[][] boardData contains String[] arrays which represent key-value pairs for the playerPositions. We will have to communicate with useCaseInteractor to determine how this information will be stored. (ex. String Player.name : Integer ) 
I will handle creation of Cells in gameCreationInteractor in a follow-up PR. Players for a Board are parsed as indicated by the playerData array above. 

Notes: 
- We may need a Player constructor which gets passed an ArrayList of Property objects to instantiate a Player who already owns Properties when loading a game. 
- I will add JavaDocs and Unit Tests in a follow-up PR to this branch

### CODE REVIEW

File: src/main/java/Interactors/TextFileTranslator.java [hunk 1]
```diff
@@ -31,33 +36,74 @@ public File[] checkSaves(String folderPath){
...
+                case "treeEnd":
+                    tree = false;
+                    break;
+                case "optionStart":
+                    options = true;
+                    break;
+                case "optionEnd":
+                    options = false;
+                    break;
             }
```
Comment (kRichard32) [inline src/main/java/Interactors/TextFileTranslator.java:85]: We can likely cut down the switch block in half by having some switches occur at the same time. However, this is slightly easier to read, so we might not want to do that.

Comment (Leuxll) [reply src/main/java/Interactors/TextFileTranslator.java:85]: Good use of the switch statements compared to the if/else statements

File: src/main/java/Interactors/TextFileTranslator.java [hunk 2]
```diff
@@ -31,33 +36,74 @@ public File[] checkSaves(String folderPath){
...
+            } else if (board) {
+                String[] position = data.trim().split(","); // denotes a key-value pair representing a Player name and their position
+                playerPositions.add(position);
+
+            } else if (tree) {
+                String[] treeData = data.trim().split(",");
+                savedTree.add(treeData);
+            } else if (options) {
+                String[] selectedOptions = data.trim().split(",");
+                treeOptions.add(selectedOptions);
```
Comment (Leuxll) [inline src/main/java/Interactors/TextFileTranslator.java:100]: Could probably use switch statements here as well, but this should also be fine

**Review by kRichard32 — APPROVED** [review]
Overall very neat and understandable code. It tackles the problem of saving and loading by separating them into basic cases. Good job.

**Review by Leuxll — APPROVED** [review]

**Review by harshkpatel — APPROVED** [review]
Overall the changes look good and another point is that you acknowledged possible issues that we need to change. I like how you outlined elements of your pr in the description as there is no javaDoc. I don't think there is any constructive feedback for any changes that need to be made right now, but I think we have a potential for expanding on this depending on the rest of the classes that need to be implemented.

**Pull request merged**